core/src/main/java/org/infinispan/protostream/impl/TagReaderImpl.java

```diff
@@ -835,12 +789,19 @@ protected void skipRawBytes(int length) throws IOException {
       }
 
       @Override
-      int setGlobalLimit(int globalLimit) {
+      public Decoder decoderFromLength(int length) {
+         ByteBuffer buffer = buf.slice().limit(length);
```

Comment:
shouldn't check if there are enough remaining bytes and throw `log.messageTruncated()` if not?